### ISSUE TITLE
[Feature]: Execute catkin_make on docker compose up

### ISSUE THREAD

**Issue by vinzenzm** [una-auxme/paf#420]
### Description

Currently it is not possible to write cpp code (Nodes or RVIZ-Panels), and be able to use it without either:
1 . rebuilding the entire agent
or 2. opening a shell with the running agent, navigating into catkin_ws and executing catkin_make manually

As I myself doesnt have to much knowledge about docker maybe an expert can give more insight about possible solutions. 

This came up in #419. 

We should take into consideration that this might not be always wanted for a speedy startup of the leaderboard simulation. 

An easy way would be to add it to "entrypoint.sh", but this might not be wanted. 

### Definition of Done

- [ ]  Thought and decided on a good solution to execute catkin_make if the user wishes
- [ ]  Implement the solution

**Comment by ll7** [comment]
Can you help me understand what is insufficient with the catkin make during the docker build phase?

https://github.com/una-auxme/paf/blob/4bea66b397f918e3eeffcdbdd616c98a6170e5c9/build/docker/agent/Dockerfile#L165-L169

**Comment by vinzenzm** [comment]
For python scripts no rebuilding is needed after a file is changed. However for cpp nodes catkin_make needs to compile the code for new changes to apply. This is important if we plan to implement our own rviz/rqt panels, such as in #419. 
Also if we want to define our own message or service types calling catkin_make is necessary.

**Comment by vinzenzm** [comment]
With 8694e5e6a82a80dbab0b587a12c974824b6f1a48 I added a vscode task to the .vscode folder which can be executed when docker container is running. 
When running the task (f1-task-runtask) the user needs to select the appropriate container by its name.
I implemented two version. One of them requires a new vs-code-extension but is able to generate the docker-container list automatically. With the second one the user needs to type in the container name, which is defaulted to 'build-agent-dev-1'. The naming however might be different on different machines. This might be uncomfortable to use. 

Is it ok to add a new Extension to the recommendations? Should we merge both versions into main, or should we decide on one option?
